error build smstools3 (use libiconv-full)

make[3]: Entering directory '/magazyn/Pliki/openwrt/trunk/feeds/packages/utils/smstools3'
. /magazyn/Pliki/openwrt/trunk/include/shell.sh; gzip -dc /magazyn/Pliki/openwrt/trunk/dl/smstools3-3.1.15.tar.gz | tar -C /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.. -xf - 

Applying ./patches/001-smsd.patch using plaintext: 
patching file src/smsd.c

Applying ./patches/002-Makefile.patch using plaintext: 
patching file src/Makefile
touch /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.prepared_cbfe09455b5053d5fe735ed375b08c37
rm -f /magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/stamp/.smstools3_installed
(cd /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/./; if [ -x ./configure ]; then find /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/ -name config.guess | xargs -r chmod u+w; find /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/ -name config.guess | xargs -r -n1 cp --remove-destination /magazyn/Pliki/openwrt/trunk/scripts/config.guess; find /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/ -name config.sub | xargs -r chmod u+w; find /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/ -name config.sub | xargs -r -n1 cp --remove-destination /magazyn/Pliki/openwrt/trunk/scripts/config.sub; AR="mips-openwrt-linux-musl-gcc-ar" AS="mips-openwrt-linux-musl-gcc -c -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64" LD=mips-openwrt-linux-musl-ld NM="mips-openwrt-linux-musl-gcc-nm" CC="mips-openwrt-linux-musl-gcc" GCC="mips-openwrt-linux-musl-gcc" CXX="mips-openwrt-linux-musl-g++" RANLIB="mips-openwrt-linux-musl-gcc-ranlib" STRIP=mips-openwrt-linux-musl-strip OBJCOPY=mips-openwrt-linux-musl-objcopy OBJDUMP=mips-openwrt-linux-musl-objdump SIZE=mips-openwrt-linux-musl-size CFLAGS="-D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64 " CXXFLAGS="-D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64 " CPPFLAGS="-I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include " LDFLAGS="-L/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/lib -znow -zrelro -liconv "   ./configure --target=mips-openwrt-linux --host=mips-openwrt-linux --build=x86_64-linux-gnu --program-prefix="" --program-suffix="" --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib --sysconfdir=/etc --datadir=/usr/share --localstatedir=/var --mandir=/usr/man --infodir=/usr/info    ; fi; )
rm -f /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.configured_*
touch /magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.configured_yyyyyyyy
make -C "/magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/src" CC="mips-openwrt-linux-musl-gcc" CFLAGS='-D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"' LFLAGS='-L/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/lib -L/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/lib -znow -zrelro -liconv  -L"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/lib"' all
make[4]: Entering directory '/magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/src'
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o extras.o extras.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o locking.o locking.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o cfgfile.o cfgfile.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o logging.o logging.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o alarm.o alarm.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o smsd_cfg.o smsd_cfg.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o charset.o charset.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
mips-openwrt-linux-musl-gcc -D NUMBER_OF_MODEMS=1 -D USE_ICONV -D DISABLE_INET_SOCKET -W -Wall -D_FILE_OFFSET_BITS=64  -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/usr/include -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include/fortify -I/magazyn/Pliki/openwrt/trunk/staging_dir/toolchain-mips_34kc_gcc-4.8-linaro_musl-1.1.11/include  -I"/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/usr/lib/libiconv-full/include"   -c -o stats.o stats.c
cc1: note: someone does not honour COPTS correctly, passed 0 times
stats.c:30:16: fatal error: mm.h: No such file or directory
 #include <mm.h>
                ^
compilation terminated.
<builtin>: recipe for target 'stats.o' failed
make[4]: *** [stats.o] Error 1
make[4]: Leaving directory '/magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/src'
Makefile:74: recipe for target '/magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.built' failed
make[3]: *** [/magazyn/Pliki/openwrt/trunk/build_dir/target-mips_34kc_musl-1.1.11/smstools3/.built] Error 2
make[3]: Leaving directory '/magazyn/Pliki/openwrt/trunk/feeds/packages/utils/smstools3'
package/Makefile:191: recipe for target 'package/feeds/packages/smstools3/compile' failed
make[2]: *** [package/feeds/packages/smstools3/compile] Error 2
make[2]: Leaving directory '/magazyn/Pliki/openwrt/trunk'
package/Makefile:188: recipe for target '/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/stamp/.package_compile' failed
make[1]: *** [/magazyn/Pliki/openwrt/trunk/staging_dir/target-mips_34kc_musl-1.1.11/stamp/.package_compile] Error 2
make[1]: Leaving directory '/magazyn/Pliki/openwrt/trunk'
/magazyn/Pliki/openwrt/trunk/include/toplevel.mk:177: polecenia dla obiektu 'world' nie powiodły się
make: *** [world] Błąd 2

--- a/utils/smstools3/Makefile
+++ b/utils/smstools3/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=smstools3
 PKG_VERSION:=3.1.15
-PKG_RELEASE:=2
+PKG_RELEASE:=3
 
 PKG_MAINTAINER:=Gérald Kerma <dreagle@doukki.net>
 PKG_LICENSE:=GPL-2.0
@@ -22,15 +22,15 @@ PKG_SOURCE_URL:=http://smstools3.kekekasvi.com/packages/
 PKG_MD5SUM:=0241ef60e646fac1a06254a848e61ed7
 
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-PKG_BUILD_DEPENDS:=libiconv-full iconv
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
 
 define Package/smstools3
   SECTION:=utils
   CATEGORY:=Utilities
   TITLE:=SMS Server Tools 3
-  DEPENDS:=+libiconv-full +iconv
+  DEPENDS:=$(ICONV_DEPENDS)
   URL:=http://smstools3.kekekasvi.com/
 endef
 
@@ -39,23 +39,17 @@ define Package/smstools3/description
  short messages through GSM modems and mobile phones.
 endef
 
-TARGET_CFLAGS = -D NUMBER_OF_MODEMS=1
+TARGET_CFLAGS += -D NUMBER_OF_MODEMS=1
 TARGET_CFLAGS += -D USE_ICONV
 TARGET_CFLAGS += -D DISABLE_INET_SOCKET
 TARGET_CFLAGS += -W -Wall
 TARGET_CFLAGS += -D_FILE_OFFSET_BITS=64
 
-TARGET_LDFLAGS += -liconv
+MAKE_VARS += LFLAGS="$(TARGET_LDFLAGS) $(if $(ICONV_FULL),-liconv)"
 
 BINDIR=/usr/local/bin
 
-define Build/Compile
-	$(MAKE) -C "$(PKG_BUILD_DIR)/src" \
-		CC="$(TARGET_CC)" \
-		CFLAGS='$(TARGET_CFLAGS) $(EXTRA_CFLAGS) $(TARGET_CPPFLAGS) $(EXTRA_CPPFLAGS) \
-		-I"$(STAGING_DIR)/usr/lib/libiconv-full/include"' \
-		LFLAGS='$(TARGET_LDFLAGS) $(EXTRA_LDFLAGS) -L"$(STAGING_DIR)/usr/lib/libiconv-full/lib"' \
-		all
+define Package/smstools3/configure
 endef
 
 define Package/smstools3/install

--- a/utils/smstools3/patches/002-Makefile.patch
+++ b/utils/smstools3/patches/002-Makefile.patch
@@ -1,6 +1,15 @@
---- a/src/Makefile	2014-12-20 18:33:55.654252867 +0100
-+++ b/src/Makefile	2014-12-20 18:31:10.241359741 +0100
-@@ -43,7 +43,7 @@
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -1,7 +1,7 @@
+ # In case of windows, use os_cygwin=yes setting in the configuration file (smsd.conf).
+ 
+ # Select your setup size:
+-CFLAGS = -D NUMBER_OF_MODEMS=64
++CFLAGS ?= -D NUMBER_OF_MODEMS=64
+ 
+ # Uncomment for Solaris
+ # CFLAGS += -D SOLARIS
+@@ -43,7 +43,7 @@ smsd: smsd.c extras.o locking.o cfgfile.
  
  ifneq (,$(findstring SOLARIS,$(CFLAGS)))
  ifeq (,$(findstring DISABLE_INET_SOCKET,$(CFLAGS)))
@@ -9,4 +18,3 @@
  endif
  endif
  
-

